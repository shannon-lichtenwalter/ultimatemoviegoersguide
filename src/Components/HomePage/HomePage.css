select {
  margin: 10px;
  border: none;
  font-size: 1.5rem; 
  background-color: transparent;
}

select:hover {
  border: 1px solid #102619;
  border-radius: 3px;
}

section {
  display: flex;
  flex-direction: column;
  align-items: center;
}
