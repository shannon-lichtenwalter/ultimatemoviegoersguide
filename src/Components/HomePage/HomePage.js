import React from 'react';
import GetMovieLists from '../../Services/getMovieLists';
import Movie from '../Movie/Movie';

class HomePage extends React.Component{
  state = {
    display: 'now_playing',
    data: null
  }

  getList = (type) => {
    GetMovieLists.getList(type)
    .then((res) => {
      this.setState({
        data:res
      })
    })
    .catch(e => this.props.setError(e));
  }

  updateMovieList = (e) => {
    this.setState({
      display:e.target.value
    }, () => this.getList(this.state.display))
  }

  componentDidMount = () => {
    if(this.props.error){
      this.props.resetError()
    }

    this.getList(this.state.display)
  }
  render(){
    return(
      <section>
        <select onChange={(e)=>this.updateMovieList(e)}>
          <option value='now_playing'>Now Playing</option>
          <option value='popular'>Popular</option>
          <option value='top_rated'>Top Rated</option>
        </select>
        <ul className='movies'>
          {this.state.data && this.state.data.results.map((movie, index) => {
            return <Movie key={index} data={movie} showDetails={this.props.showDetails}/>
          })}
        </ul>
      </section>
    )
  }
}

export default HomePage;