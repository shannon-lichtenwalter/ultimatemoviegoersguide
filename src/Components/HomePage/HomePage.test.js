import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from './HomePage';

describe('HomePage Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<HomePage />, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the HomePage correctly when rendered', () => {
    const wrapper = mount(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

});
