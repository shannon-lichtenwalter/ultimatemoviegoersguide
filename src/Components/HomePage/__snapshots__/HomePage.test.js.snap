// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component Displays the HomePage correctly when rendered 1`] = `
<HomePage>
  <section>
    <form>
      <label
        hidden={true}
        htmlFor="listSelect"
      >
        Displaying:
      </label>
      <select
        id="listSelect"
        onChange={[Function]}
      >
        <option
          value="now_playing"
        >
          Now Playing
        </option>
        <option
          value="popular"
        >
          Popular
        </option>
        <option
          value="top_rated"
        >
          Top Rated
        </option>
      </select>
    </form>
    <ul
      className="movies"
    />
  </section>
</HomePage>
`;
